cpp/include/cudf/stream_compaction.hpp

```diff
@@ -421,5 +422,47 @@ cudf::size_type distinct_count(table_view const& input,
                                null_equality nulls_equal    = null_equality::EQUAL,
                                rmm::cuda_stream_view stream = cudf::get_default_stream());
 
+/**
+ * @brief Creates a new column by applying a filter function against every
+ * element of the input columns.
+ *
+ * Null values in the input columns are considered as not matching the filter.
+ *
+ * Computes:
+ * `out[i]... = predicate(columns[i]... ) ? (columns[i]...): not-applied`.
+ *
+ * Note that for every scalar in `columns` (columns of size 1), `columns[i] ==
+ * input[0]`
+ *
+ *
+ * @throws std::invalid_argument if any of the input columns have different sizes (except scalars of
+ * size 1)
+ * @throws std::invalid_argument if `output_type` or any of the inputs are not fixed-width or string
+ * types
+ * @throws std::invalid_argument if any of the input columns have nulls
+ * @throws std::logic_error if JIT is not supported by the runtime
+ * @throws std::logic_error if the size of the copy_mask does not match the size of the input
```

Comment:
```suggestion
 * @throws cudf::logic_error if JIT is not supported by the runtime
 * @throws cudf::logic_error if the size of the copy_mask does not match the size of the input
```

Reply:
Should we have tests for these exceptions?

Reply:
I've added them. thanks!